oauth2_http/java/com/google/auth/oauth2/SecureSessionAgent.java

```diff
@@ -72,23 +72,24 @@ public final class S2A {
   private static final String MDS_MTLS_ENDPOINT =
       ComputeEngineCredentials.getMetadataServerUrl() + S2A_CONFIG_ENDPOINT_POSTFIX;
 
-  private S2AConfig config;
-
   private transient HttpTransportFactory transportFactory;
 
-  S2A(S2A.Builder builder) {
+  SecureSessionAgent(SecureSessionAgent.Builder builder) {
     this.transportFactory = builder.getHttpTransportFactory();
-    this.config = getS2AConfigFromMDS();
   }
 
-  /** @return the mTLS S2A Address from the mTLS config. */
-  public String getMtlsS2AAddress() {
-    return config.getMtlsAddress();
+  /**
+   * This method makes a network call to MDS to get the {@link SecureSessionAgentConfig} which
+   * contains the plaintext and mtls address to reach the S2A (Secure Session Agent).
+   *
+   * @return a SecureSessionAgentConfig.
+   */
+  public SecureSessionAgentConfig getConfig() {
+    return getSecureSessionAgentConfigFromMDS();
   }
 
-  /** @return the plaintext S2A Address from the mTLS config. */
-  public String getPlaintextS2AAddress() {
-    return config.getPlaintextAddress();
+  public static SecureSessionAgent create() {
+    return newBuilder().build();
```

Comment:
```suggestion
  /** @return default instance of SecureSessionAgent */
  public static SecureSessionAgent create() {
    return newBuilder().build();
```

or something similar